jbpm-flow/src/main/java/org/jbpm/ruleflow/instance/RuleFlowProcessInstanceFactory.java

```diff
@@ -23,13 +25,51 @@
 
 import org.jbpm.process.instance.AbstractProcessInstanceFactory;
 import org.jbpm.process.instance.ProcessInstance;
+import org.kie.api.runtime.Environment;
 
 public class RuleFlowProcessInstanceFactory extends AbstractProcessInstanceFactory implements Externalizable {
 
     private static final long serialVersionUID = 510l;
 
-    public ProcessInstance createProcessInstance() {
-        return new RuleFlowProcessInstance();
+    // OCRAM: rename to "queue-based"
```

Comment:
MARCO backwards: it's the tag I used to mark places that I need to keep track of and that are important.